File: src/3300.minimum-element-after-replacement-with-digit-sum.3606/solution.ts
function minElement (nums: number[]): number {
    return Math.min(...nums.map(n => {
        let sum = 0;
        while (n > 0) {
            const digit = n % 10;
            sum += digit;
            n = (n - digit) / 10;
        }
        return sum;
    }));
}

File: src/3300.minimum-element-after-replacement-with-digit-sum.3606/solution.java
class Solution {
    public int minElement(int[] nums) {
        int result = Integer.MAX_VALUE;
        for(int n : nums){
            result = Math.min(result,calc(n));
        }
        return result;
    }

    private int calc(int num){
        int n = 0;
        while (num != 0){
            n += num%10;
            num /= 10;
        }
        return n;
    }
}